datamade/committee-oversight
# Add committee detail page

## Overview

Closes #66 

This PR:

- Refactors common detail page functionality into an abstract `DetailPage` base class, and uses this base class to add a `CommitteeDetailPage`.
- Adds a `committee_detail_page.html` template.
- Enables, but hides, the Django admin interfaces for `CommitteeOrganization` and `CommitteeDetailPages` to enable autocomplete and take advantage of the autocomplete widget on the `CommitteeDetailPage` add and edit screens.
- Overrides the Javascript blocks in the edit and create Wagtail templates to put vendor Javascript before Wagtail Javascript. This allows us to hook into Django's Select2 Javascript, which was otherwise impossible, in order to dynamically set the slug and title on instances of `CommitteeDetailPage`.
- Adds a starter template for committee detail pages.

## Testing Instructions

- With the app running, run the migrations: `docker-compose exec app python manage.py migrate`
- In the Wagtail admin, navigate to Pages > The Committee Oversight Data Project.
- Click Add child page, and select Committee detail page.
- Publish a committee page or two. Confirm that the slug and title are updated as expected in the Promote tab.
- Back in the Pages listing, click the "Live" link to the far right to view your new detail pages. Confirm that the starter template contains the correct information and looks nice (enough).
